pkg/kapis/openpitrix/v1/handler.go

```diff
@@ -44,28 +57,8 @@ func (h *openpitrixHandler) ListApplications(request *restful.Request, response
 		api.HandleBadRequest(response, nil, err)
 		return
 	}
-
-	// filter namespaced applications by runtime_id
-	if namespace != "" {
-		ns, err := h.informers.Core().V1().Namespaces().Lister().Get(namespace)
-
-		if err != nil {
-			klog.Errorln(err)
-			api.HandleInternalError(response, nil, err)
-			return
-		}
-
-		runtimeId := ns.Annotations[constants.OpenPitrixRuntimeAnnotationKey]
-
-		if runtimeId == "" {
-			// runtime id not exist,return empty response
-			response.WriteAsJson(models.PageableResponse{Items: []interface{}{}, TotalCount: 0})
-			return
-		} else {
-			// filter by runtime id
-			conditions.Match[openpitrix.RuntimeId] = runtimeId
-		}
-	}
+	conditions.Match[openpitrix.Zone] = namespace
+	conditions.Match[openpitrix.RuntimeId] = runtimeId
 
 	result, err := h.openpitrix.ListApplications(conditions, limit, offset, orderBy, reverse)
```

Comment:
should handle the zone condition in `pkg/models/openpitrix/applications.go:78 ListApplications `